src/libbson/src/bson/bson-dsl.md

```diff
@@ -0,0 +1,1029 @@
+calling thread. **NOTE:** It is the caller's responsibility to eventually free
+`P`.
+
+
+#### `if()`
+
+> `if(Predicate cond, then(VisitOperation... thens))`
+> `if(Predicate cond, then(VisitOperation... thens), else(VisitOperation... elses))`
+
+If the *Predicate* `cond` matches the current element, apply the `thens`
+operations, otherwise the `elses` (if given).
+
+**NOTE** that this `if()` command expects a DSL *Predicate* as its condition,
+and *not* a regular C expression like the other `if()` DSL commands.
+
+
+#### `nop`
+
+Does nothing.
+
+
+#### `case()`
+
+> ```
+> case(when(Predicate w,
+>           VisitOperations... ops)
+>      ...,
+>      [else(VisitOperations... ops)])
+> ```
+
+Perform case selection based on a set of predicates.
+
+`case` accepts zero or more `when(Predicate w, VisitOperation... ops)` clauses.
+Each `when` is considered in the listed order. If the *Predicate* `w` of a
+`when` matches the `bsonVisitIter` element, then the `when`'s *VisitOperations*
+`ops` are evaluated on the element.
+
+After any `when` clause matches an element, no subsequent `when` clauses are
+considered.
+
+An `else(ops...)` clause is equivalent to `when(true, ops...)`.
+
+If no clause matches the element, nothing happens.
+
+
+### *Predicate*
+
+DSL *Predicates* appears in several locations. A *Predicate* evaluates some
+condition on the current `bsonVisitIter` (which is updated automatically by
+other DSL commands). A predicate can be evaluated at any point in C code by
+using the `bsonPredicate()` expression (this is only valid within a `visitEach`
+or `parse` context, since it references the `bsonVisitIter` name).
+
+The following predicates are defined:
+
+
+#### `key(const char* s...)`
+
+Specify one or more C strings for `s`. This predicate will match if *any of* `s`
+is equivalent to the element's key string.
+
+The `iKey(s...)` predicate is the same as `key`, but performs a case-insensitive
+match.
+
+
+#### `type(TypeName t)`
+
+Matches if the element's current type matches the `TypeName` `t`.
+
+The `TypeName`s are: `double`, `utf8`, `doc`, `array`, `binary`, `undefined`,
+`oid`, `bool`, `date_time`, `null`, `regex`, `dbpointer`, `code`, `codewscope`,
+`int32`, `timestamp`, `int64`, and `decimal128`.
+
+
+#### `keyWithType(const char* k, TypeName t)`
+
+Matches an element with key `k` and type `t`.
+
+`iKeyWithType` performs a case-insensitive string match for the key.
+
+
+#### `isNumeric`
+
+The `isNumeric` predicate matches if the element's value is a numeric type
+(`double`, `int32`, or `int64`. `decimal128` is *not* included).
+
+
+#### `lastElement`
+
+Matches only the final element in a document.
+
+
+#### `true`
+
+Always matches.
+
+
+#### `false`
+
+Never matches.
+
+
+#### `truthy`
+
+Matches is the element is "truthy": A `true` boolean, a non-zero numeric type,
+or other non-null value.
+
+
+#### `falsey`
+
+Matches any element that is not `truthy`
```

Comment:
```suggestion
#### `is_true`

Matches if the element is `true` according to `bson_iter_as_bool`.


#### `is_false`

Matches if the element is `false` according to `bson_iter_as_bool`.
```

Suggest renaming `truthy`/`falsey` to `is_true` and `is_false` and adjusting the description to be defined in terms of `bson_iter_as_bool`.

Reply:
Renamed with `isTrue` and `isFalse` (to match the casing of everything else)